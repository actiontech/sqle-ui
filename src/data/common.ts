export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}