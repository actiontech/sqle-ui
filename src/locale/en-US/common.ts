// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: 'username',
  password: 'password',

  unknownError: 'Unknown Error',

  ok: 'Ok',
  cancel: 'Cancel',
  submit: 'Submit',
  close: 'Close',
  edit: 'Edit',
  delete: 'Delete',
  reset: 'Reset',
  search: 'Search',
  retry: 'Retry',
  back: 'Back',
  upload: 'Upload',
  resetAndClose: 'Reset Form And Close',
  operateSuccess: 'Operate successfully',
  prevStep: 'Prev',
  nextStep: 'Next',

  expansion: 'Expand',
  collapse: 'Collapse',

  theme: {
    light: 'Light Theme',
    dark: 'Dark Theme',
  },

  logout: 'Logout',
  account: 'Account',

  nav: {
    title: 'SQLe',
  },

  request: {
    noticeFailTitle: 'Request Failed',
  },

  form: {
    placeholder: {
      input: 'Please input {{name}}',
      select: 'Please select {{name}}',
      searchInput: 'Input the {{name}} to search',
      searchSelect: 'Select the {{name}} to search',
    },
    rule: {
      require: 'Please input {{name}}',
      passwordNotMatch: 'The two passwords that you entered do not match!',
      email: 'The input is not valid E-mail!',
    },
  },
};
