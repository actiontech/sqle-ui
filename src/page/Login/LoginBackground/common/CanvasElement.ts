import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
