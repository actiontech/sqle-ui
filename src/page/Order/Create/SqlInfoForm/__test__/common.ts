import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
