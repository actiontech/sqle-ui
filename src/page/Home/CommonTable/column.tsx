import { TableColumn } from '../../../types/common.type';
import { t } from '../../../locale';
import { formatTime } from '../../../utils/Common';
import { IWorkflowDetailResV1 } from '../../../api/common';
import EmptyBox from '../../../components/EmptyBox';
import { Link } from '../../../components/Link';
import { renderOrderDesc } from '../../Order/List/column';

export const commonColumn: () => TableColumn<IWorkflowDetailResV1> = () => {
  const column: TableColumn<IWorkflowDetailResV1> = [
    {
      dataIndex: 'workflow_name',
      title: () => t('order.order.name'),
      render: (text, record) => {
        return (
          <EmptyBox if={!!text && !!record.project_name} defaultNode={text}>
            <Link
              to={`project/${record.project_name}/order/${record.workflow_id}`}
            >
              {text}
            </Link>
          </EmptyBox>
        );
      },
      width: 400,
    },
    {
      dataIndex: 'project_name',
      title: () => t('projectManage.projectForm.projectName'),
      width: 300,
      render: (projectName) => {
        return (
          <EmptyBox if={!!projectName} defaultNode={projectName}>
            <Link to={`project/${projectName}/overview`}>{projectName}</Link>
          </EmptyBox>
        );
      },
    },
    {
      dataIndex: 'desc',
      title: () => t('order.order.desc'),
      width: 600,
      render: (desc: string) => renderOrderDesc(desc, 600),
    },
    {
      dataIndex: 'create_time',
      title: () => t('order.order.createTime'),
      render: (time) => {
        return formatTime(time);
      },
      width: 'auto',
    },
  ];

  return column;
};
