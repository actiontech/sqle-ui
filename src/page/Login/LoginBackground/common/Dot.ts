import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
