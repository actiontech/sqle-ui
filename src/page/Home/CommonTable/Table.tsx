import { Result, Table } from 'antd';
import { useTranslation } from 'react-i18next';
import { ICommonTableProps } from '.';
import { commonColumn } from './column';

const CommonTable: React.FC<ICommonTableProps> = ({
  tableInfo,
  customColumn,
}) => {
  const { t } = useTranslation();

  return (
    <>
      <Table
        rowKey="subject"
        bordered
        loading={tableInfo.loading}
        scroll={{ x: 'max-content' }}
        pagination={false}
        locale={{
          emptyText: tableInfo.error ? (
            <Result
              status="error"
              title={t('common.request.noticeFailTitle')}
              subTitle={tableInfo.error?.message ?? t('common.unknownError')}
            />
          ) : undefined,
        }}
        dataSource={tableInfo.data ?? []}
        columns={
          typeof customColumn === 'function' ? customColumn() : commonColumn()
        }
      />
    </>
  );
};

export default CommonTable;
