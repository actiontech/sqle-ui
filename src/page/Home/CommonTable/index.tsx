import { Badge } from 'antd';
import { IWorkflowDetailResV2 } from '../../../api/common';
import { TableColumn } from '../../../types/common.type';
import CommonTable from './Table';

export const genTabPaneTitle = (title: string, badgeCount?: number) => {
  return badgeCount ? (
    <Badge
      className="tabs-panel-badge"
      count={badgeCount}
      overflowCount={99}
      size="small"
    >
      <span className="tabs-panel-title">{title}</span>
    </Badge>
  ) : (
    title
  );
};

export type CommonTableInfoType = {
  data: IWorkflowDetailResV2[];
  error: Error | undefined;
  loading: boolean;
};

export interface ICommonTableProps {
  tableInfo: CommonTableInfoType;
  customColumn?: () => TableColumn<IWorkflowDetailResV2>;
}

export const DASHBOARD_COMMON_GET_ORDER_NUMBER = 5;

export default CommonTable;
