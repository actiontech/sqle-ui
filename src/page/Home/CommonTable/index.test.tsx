import { render, screen } from '@testing-library/react';
import { IWorkflowDetailResV1 } from '../../../api/common';
import CommonTable, { CommonTableInfoType, genTabPaneTitle } from './index';
import { renderWithRouter } from '../../../testUtils/customRender';
import { TableColumn } from '../../../types/common.type';
import { getHrefByText } from '../../../testUtils/customQuery';

const list: IWorkflowDetailResV1[] = [
  {
    workflow_id: '1',
    workflow_name: 'test1',
    project_name: 'default',
    desc: 'desc1',
    create_time: '',
  },
];

let mockTableInfo: CommonTableInfoType = {
  loading: false,
  data: list,
  error: undefined,
};

describe('test Home/genTabPaneTitle', () => {
  test('should match snapshot', () => {
    expect(genTabPaneTitle('title')).toMatchSnapshot();
    expect(genTabPaneTitle('title', 12)).toMatchSnapshot();
    expect(genTabPaneTitle('title', 101)).toMatchSnapshot();
  });

  test('should only render the title when badgeCount is zero or undefined', () => {
    const Element = genTabPaneTitle('title');
    render(Element as JSX.Element);
    expect(screen.getByText('title')).toBeInTheDocument();
    expect(document.querySelector('.tabs-panel-badge')).toBeNull();
  });

  test('should be rendered badge count when badgeCount is not zero or has been defined', () => {
    const Element = genTabPaneTitle('title', 88);
    render(Element as JSX.Element);
    expect(document.querySelector('.tabs-panel-badge')).not.toBeNull();
    expect(Element).toMatchSnapshot();
  });

  test('should be rendered 99+ when badgeCount number is greater than 99', () => {
    const Element = genTabPaneTitle('title', 128);
    render(Element as JSX.Element);
    expect(screen.getByText('99+')).toBeInTheDocument();
  });
});

describe('test Home/CommonTable', () => {
  afterEach(() => {
    mockTableInfo = {
      loading: false,
      data: list,
      error: undefined,
    };

    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = renderWithRouter(
      <CommonTable tableInfo={mockTableInfo} />
    );
    expect(container).toMatchSnapshot();
  });

  test('should render spin when loading equal true', () => {
    mockTableInfo.loading = true;
    const { container } = renderWithRouter(
      <CommonTable tableInfo={mockTableInfo} />
    );
    expect(container).toMatchSnapshot();
  });

  test.skip('should render error info when error has a value', () => {
    mockTableInfo.error = new Error('error message');
    mockTableInfo.data = [];
    const { container } = renderWithRouter(
      <CommonTable tableInfo={mockTableInfo} />
    );

    expect(container).toMatchSnapshot();
    expect(
      screen.getByText('common.request.noticeFailTitle')
    ).toBeInTheDocument();
    expect(screen.getByText('error message')).toBeInTheDocument();
  });

  test('should render custom custom when custom has a value', () => {
    const mockCustomColumn = jest
      .fn<TableColumn<IWorkflowDetailResV1>, any>()
      .mockReturnValue([
        {
          dataIndex: 'workflow_name',
          title: () => 'subject',
        },
        {
          dataIndex: 'desc',
          title: () => 'desc',
        },
        {
          dataIndex: 'workflow_name',
          title: () => 'workflow_id',
        },
      ]);
    const { container, rerender } = renderWithRouter(
      <CommonTable tableInfo={mockTableInfo} />
    );

    expect(mockCustomColumn).toBeCalledTimes(0);

    rerender(
      <CommonTable tableInfo={mockTableInfo} customColumn={mockCustomColumn} />
    );
    expect(mockCustomColumn).toBeCalledTimes(1);
    expect(container).toMatchSnapshot();
  });

  test('should jump to the order page under the project when click on the corresponding link', () => {
    renderWithRouter(<CommonTable tableInfo={mockTableInfo} />);

    expect(screen.getByText(list[0].workflow_name!)).toBeInTheDocument();
    expect(getHrefByText(list[0].workflow_name!)).toBe(
      `/project/${list[0].project_name}/order/${list[0].workflow_id}`
    );

    expect(screen.getByText(list[0].project_name!)).toBeInTheDocument();
    expect(getHrefByText(list[0].project_name!)).toBe(
      `/project/${list[0].project_name}/overview`
    );
  });
});
