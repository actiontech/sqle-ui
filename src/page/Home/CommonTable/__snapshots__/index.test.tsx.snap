// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Home/CommonTable should match snapshot 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-bordered ant-table-scroll-horizontal"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
              style="overflow-x: auto; overflow-y: hidden;"
            >
              <table
                style="min-width: 100%; table-layout: auto;"
              >
                <colgroup>
                  <col />
                  <col />
                  <col
                    style="width: 30%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell"
                    >
                      order.order.name
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      projectManage.projectForm.projectName
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      order.order.desc
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      order.order.createTime
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    aria-hidden="true"
                    class="ant-table-measure-row"
                    style="height: 0px; font-size: 0px;"
                  >
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="1"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      <a
                        href="/project/default/order/1"
                      >
                        test1
                      </a>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <a
                        href="/project/default/overview"
                      >
                        default
                      </a>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line margin-bottom-0"
                        style="max-width: 600px;"
                      >
                        desc1
                      </div>
                    </td>
                    <td
                      class="ant-table-cell"
                    />
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Home/CommonTable should render custom custom when custom has a value 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-bordered ant-table-scroll-horizontal"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
              style="overflow-x: auto; overflow-y: hidden;"
            >
              <table
                style="min-width: 100%; table-layout: auto;"
              >
                <colgroup />
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell"
                    >
                      subject
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      desc
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      workflow_id
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    aria-hidden="true"
                    class="ant-table-measure-row"
                    style="height: 0px; font-size: 0px;"
                  >
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="1"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      test1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      desc1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      test1
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Home/CommonTable should render error info when error has a value 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-bordered ant-table-empty ant-table-scroll-horizontal"
      >
        <div
          class="ant-table-container"
        >
          <div
            class="ant-table-content"
            style="overflow-x: auto; overflow-y: hidden;"
          >
            <table
              style="min-width: 100%; table-layout: auto;"
            >
              <colgroup>
                <col />
                <col />
                <col
                  style="width: 30%;"
                />
              </colgroup>
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-cell"
                  >
                    order.order.name
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    projectManage.projectForm.projectName
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    order.order.desc
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    order.order.createTime
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  aria-hidden="true"
                  class="ant-table-measure-row"
                  style="height: 0px; font-size: 0px;"
                >
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-table-placeholder"
                >
                  <td
                    class="ant-table-cell"
                    colspan="4"
                  >
                    <div
                      class="ant-table-expanded-row-fixed"
                      style="width: 0px; position: sticky; left: 0px; overflow: hidden;"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Home/CommonTable should render spin when loading equal true 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div>
        <div
          aria-busy="true"
          aria-live="polite"
          class="ant-spin ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <div
        class="ant-spin-container ant-spin-blur"
      >
        <div
          class="ant-table ant-table-bordered ant-table-scroll-horizontal"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
              style="overflow-x: auto; overflow-y: hidden;"
            >
              <table
                style="min-width: 100%; table-layout: auto;"
              >
                <colgroup>
                  <col />
                  <col />
                  <col
                    style="width: 30%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell"
                    >
                      order.order.name
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      projectManage.projectForm.projectName
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      order.order.desc
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      order.order.createTime
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    aria-hidden="true"
                    class="ant-table-measure-row"
                    style="height: 0px; font-size: 0px;"
                  >
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                    <td
                      style="padding: 0px; border: 0px; height: 0px;"
                    >
                      <div
                        style="height: 0px; overflow: hidden;"
                      >
                         
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="1"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      <a
                        href="/project/default/order/1"
                      >
                        test1
                      </a>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <a
                        href="/project/default/overview"
                      >
                        default
                      </a>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line margin-bottom-0"
                        style="max-width: 600px;"
                      >
                        desc1
                      </div>
                    </td>
                    <td
                      class="ant-table-cell"
                    />
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Home/genTabPaneTitle should be rendered badge count when badgeCount is not zero or has been defined 1`] = `
<Badge
  className="tabs-panel-badge"
  count={88}
  overflowCount={99}
  size="small"
>
  <span
    className="tabs-panel-title"
  >
    title
  </span>
</Badge>
`;

exports[`test Home/genTabPaneTitle should match snapshot 1`] = `"title"`;

exports[`test Home/genTabPaneTitle should match snapshot 2`] = `
<Badge
  className="tabs-panel-badge"
  count={12}
  overflowCount={99}
  size="small"
>
  <span
    className="tabs-panel-title"
  >
    title
  </span>
</Badge>
`;

exports[`test Home/genTabPaneTitle should match snapshot 3`] = `
<Badge
  className="tabs-panel-badge"
  count={101}
  overflowCount={99}
  size="small"
>
  <span
    className="tabs-panel-title"
  >
    title
  </span>
</Badge>
`;
